google/cloud/storage/tests/testbench.py

```diff
@@ -146,13 +169,52 @@ def __init__(self, gcs_url, bucket_name, name, generation, request):
             'size': len(self.media),
             'etag': 'XYZ='
         }
+        self._capture_customer_encryption(request)
+
         self.insert_acl(
             canonical_entity_name('project-owners-123456789'), 'OWNER')
         self.insert_acl(
             canonical_entity_name('project-editors-123456789'), 'OWNER')
         self.insert_acl(
             canonical_entity_name('project-viewers-123456789'), 'READER')
 
+    def _capture_customer_encryption(self, request):
+        """Capture the customer-supplied encryption key, if any.
+
+        :param request:flask.Request the http request.
+        :return:NoneType
+        """
+        if request.headers.get('X-Goog-Encryption-Key') is None:
```

Comment:
Also change header names to all-lowercase here and below.

Reply:
Done.